prow/git/v2/adapter.go

```diff
@@ -0,0 +1,69 @@
+package git
+
+import (
+	"errors"
+
+	"k8s.io/test-infra/prow/git"
+	"k8s.io/test-infra/prow/github"
+)
+
+// ClientFactoryFrom adapts the v1 client to a v2 client
+func ClientFactoryFrom(c *git.Client) ClientFactory {
+	return &clientFactoryAdapter{Client: c}
+}
+
+type clientFactoryAdapter struct {
+	*git.Client
+}
+
+// ClientFromDir creates a client that operates on a repo that has already
+// been cloned to the given directory.
+func (a *clientFactoryAdapter) ClientFromDir(org, repo, dir string) (RepoClient, error) {
+	return nil, errors.New("no ClientFromDir implementation exists in the v1 git client")
+}
+
+// Repo creates a client that operates on a new clone of the repo.
+func (a *clientFactoryAdapter) ClientFor(org, repo string) (RepoClient, error) {
+	r, err := a.Client.Clone(org, repo)
+	return &repoClientAdapter{Repo: r}, err
+}
+
+type repoClientAdapter struct {
```

Comment:
Do you want to do the:

```golang
func init() {
  var _ oldInterface = repoClientAdapter
}
```

to ensure it implements the interface correctly?

Reply:
That's implicit in the call to `NewClients()` since I return the wrapper